Sources/ScipioKit/Producer/PIF/PIFCompiler.swift

```diff
@@ -18,27 +19,29 @@ struct PIFCompiler: Compiler {
         buildOptions: BuildOptions,
         buildOptionsMatrix: [String: BuildOptions],
         fileSystem: any FileSystem = TSCBasic.localFileSystem,
-        executor: any Executor = ProcessExecutor()
+        executor: any Executor = ProcessExecutor(),
+        toolchainEnvironment: [String: String]? = nil
     ) {
         self.descriptionPackage = descriptionPackage
         self.buildOptions = buildOptions
         self.buildOptionsMatrix = buildOptionsMatrix
         self.fileSystem = fileSystem
         self.executor = executor
         self.buildParametersGenerator = .init(buildOptions: buildOptions, fileSystem: fileSystem)
+        self.toolchainEnvironment = toolchainEnvironment
     }
 
     private func fetchDefaultToolchainBinPath() async throws -> AbsolutePath {
         let result = try await executor.execute("/usr/bin/xcrun", "xcode-select", "-p")
-        let rawString = try result.unwrapOutput()
+        let rawString = try result.unwrapOutput().trimmingCharacters(in: .whitespacesAndNewlines)
         let developerDirPath = try AbsolutePath(validating: rawString)
         let toolchainPath = try RelativePath(validating: "./Toolchains/XcodeDefault.xctoolchain/usr/bin")
         return developerDirPath.appending(toolchainPath)
```

Comment:
@giginet some context: without this change, the `toolchainPath` was interpolated as `/Applicaitons/Xcode.app/Developer/Contents/.\nToolchains/XcodeDefault.xctoolchain/usr/bin`. It has worked so far because SwiftPM down the road defaulted to the correct toolchain path, in the end. However, when run with different environments within the same runner it led to incorrect toolchain path determination. 

I added this change ad-hoc but it might make sense to trim whitespaces and new lines inside of the `unwrapOutput()` function so potential issues are also addressed in other places. I can do it withing this PR or you can just take it FYI.

Reply:
This change will not affect other implementations. Thank you.